### PR TITLE
Add new required `module_source_regex` to provider config, add new required `module_source` tag to `modtm_telemetry` resource

### PR DESCRIPTION
**Pull request by lonegunmanb**
Only telemetry with tag module_source that matches at least one of module_source_regex would be collected

### CODE REVIEW

**Review by matt-FFFFFF — CHANGES_REQUESTED** [review]

File: internal/provider/telemetry_resource.go [hunk 1]
```diff
@@ -264,6 +267,20 @@ func (r *TelemetryResourceModel) sendTags(ctx context.Context, res *TelemetryRes
...
+	}
+	match := false
+	for _, regex := range res.moduleSourceRegex {
+		if regex.MatchString(src) {
+			match = true
+			break
+		}
+	}
+	if !match {
+		return
```
[the remark this answers is not in the document]
Comment (lonegunmanb) [reply internal/provider/telemetry_resource.go:282]: Is a telemetry event without source helpful to the telemetry consumer? @matt-FFFFFF

[the remark this answers is not in the document]
Comment (matt-FFFFFF) [reply internal/provider/telemetry_resource.go:282]: Good point, yes I wasn't thinking clearly.

**Review by matt-FFFFFF — APPROVED** [review]

File: internal/provider/telemetry_resource.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaredfholgate) [reply internal/provider/telemetry_resource.go:282]: Can't we still get the module name in this scenario and may categorise as `inner sourced`? I think if we can still get some data from it, that would be useful to understand the percentage of people forking our modules. E.g. the categories could be:

- `inner_source_local_file_system`
- `inner_source_git`
- `inner_source_other`

Quite possible I am missing something here, so apologies if I am am, but thought we still had the module name and version available maybe???

[the remark this answers is not in the document]
Comment (jaredfholgate) [reply internal/provider/telemetry_resource.go:282]: If you wanted to be super clever, you could extract and hash the first portion of the source uri and send that too, so we can tell if it is the same source. I think that might be verging on data we shouldn't store though?

**Pull request merged**